pkg/watermark/wmb/interface.go

```diff
@@ -0,0 +1,12 @@
+package wmb
+
+import "github.com/numaproj/numaflow/pkg/isb"
+
+// IdleManagerInterface decides when to send a control message and also keeps track of idle watermark's offset.
+type IdleManagerInterface interface {
+	// Validate validates whether to send a control message for the given partition or not
+	Validate(toBufferPartitionName string) bool
```

Comment:
Naming..

`Validate` is a verb, using something like `IsNeeded`, `IsValid`, figure out a better name.

Reply:
Updated to `NeedToSendCtrlMsg`... any other suggestions.... ?...